### PR TITLE
Only log errors on server errors

### PR DESCRIPTION
**Pull request by sethvargo**
This guards entry into MakeRevisionToken against user error, such that if MakeRevisionToken returns an error, it's a server-side error.

Previously, we would log errors if a client was attempting to revise no keys, which isn't necessarily an error. There's no client-facing behavior change here, just reducing noise in the server logs.

Part of https://github.com/google/exposure-notifications-server/issues/1412

**Release Note**

```release-note
Only log server-side errors when creating a revision token and return a 500 response when making a revision token fails (previously this was undefined behavior).
```

### CODE REVIEW

File: internal/publish/publish.go [hunk 1]
```diff
@@ -450,12 +450,23 @@ func (s *Server) process(ctx context.Context, data *verifyapi.Publish, platform
...
-		logger.Errorf("unable to make new revision token: %v", err)
-		newToken = make([]byte, 0)
+	newToken := make([]byte, 0)
+	if len(keep.RevisableKeys) != 0 && len(resp.Exposures) != 0 {
+		var err error
+		newToken, err = s.tokenManager.MakeRevisionToken(ctx, &keep, resp.Exposures, s.tokenAAD)
+		if err != nil {
+			// Something failed with the revision token generation or encryption.
+			// Return an internal error so clients retry.
+			logger.Errorw("failed to make updated revision token", "error", err)
```
[the remark this answers is not in the document]
Comment (mikehelmick) [reply internal/publish/publish.go:460]: this is a client API behiavior change

in the previous version - a failed revision token still allowed the upload to move forward.

**Review by mikehelmick — CHANGES_REQUESTED** [review]
let's discus the behavior change

**Comment by sethvargo** [comment]
@mikehelmick updated

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**